README.md

```diff
@@ -1 +1,86 @@
-# iagora_deployment
\ No newline at end of file
+# iagora_deployment
+
+
+
+## How to use vm bicep deployment templates
+
+To deploy the virtual machine and associated resources using the provided Bicep template, follow these steps:
+
+### Step 1: Install Azure CLI
+
+Ensure that the Azure CLI is installed. If it is not installed, use the following command:
+
+```bash
+curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+```
+
+### Step 2: Log in to Your Azure Account
+
+After installing Azure CLI, log in to the Azure account:
+
+```bash
+az login
+```
+
+This command will open a browser window to log in with Azure credentials. If using a remote machine without a graphical interface, use:
+
+```bash
+az login --use-device-code
+```
+
+### Step 3: Install Bicep CLI
+
+Once logged in, install Bicep CLI by running:
+
+```bash
+az bicep install
+```
+
+### Step 4: Upgrade Bicep CLI (Optional)
+
+To ensure the latest version of Bicep is installed, upgrade it using:
+
+```bash
+az bicep upgrade
+```
+
+### Step 5: Verify Bicep Installation
+
+Verify the Bicep installation by checking the version:
+
+```bash
+az bicep version
+```
+
+### Step 6: (Optional) Convert Existing ARM Template to Bicep
+
+To convert an ARM template in JSON format to Bicep, use:
+
+```bash
+az bicep decompile -f path/to/ARM/template.json
+```
+
+### Step 7: Deploy the Bicep Template
+
+Since all necessary parameters are included directly within the `vm-deploy.bicep` file, there is no need to manage a separate `parameters.json` file. Deploy the resources using the following command:
+
+```bash
+az deployment group create --resource-group <your-resource-group> --template-file vm-deploy.bicep
+```
+
+**Explanation of Parameters**:
+- **`<your-resource-group>`**: Replace with the name of the Azure resource group where the resources will be deployed.
+- **`vm-deploy.bicep`**: This is the Bicep file containing the deployment configuration, including all required parameters. The `parameters.json` file has been removed as parameters are defined directly within this file.
+
+### Step 8: Customize the Deployment
+
+Parameters can be adjusted directly in the `vm-deploy.bicep` file before running the deployment command. This approach simplifies management and customization of deployments without the need for a separate parameters file.
```

Comment:
```suggestion
```
To customize parameters, we could use the `--parameters` CLI option instead; see the suggestions above.